code-challenges.js

```diff
@@ -18,9 +24,55 @@ const fruit3 = "cherry"
 const fruit4 = "kiwi"
 // Expected outcome: "cherry"
 
+/**
+ * Determines which out of two strings is longer
+ * @param {string} str1 - A string param
+ * @param {string} str2 - A string to compare
+ * @return {string} Returns the string with the longest length
+ * 
+ * @example
+ * longestString("abc", "1234")
+ * 
+ * @throws {TypeError} parameter is not a string
+ */
+const longestString = (str1, str2) => {
+    // check parameter types to ensure two strings are given
+    if (typeof str1 !== 'string') {
+        throw new TypeError('str1 is not a string')
+    } else if (typeof str2 !== 'string') {
+        throw new TypeError('str2 is not a string')
+    }
+
+    // if str1 is longer then return str1
+    if (str1.length > str2.length) {
+        return str1
+    }
+
+    // since condition above failed str2 is longer
+    return str2
+}
+
+console.log(longestString(fruit1, fruit2)) // --> banana
+console.log(longestString(fruit3, fruit4)) // --> cherry
+
 // --------------------1) Create a function that takes in a number and determines if the given number is below boiling point, at boiling point, or above boiling point. Boiling point is 212 degrees Fahrenheit. Use the test variables provided below.
 
 // Pseudo code:
+/*
+Create a function called `boilingPoint` which accepts a number parameter (temp) and a string of 'C' or 'F' or 'K' (unit) to set temperature unit (defaults to 'F')
+if unit = 'C'
+    if temp > 100 then return 'above boiling point'
+    if temp === 100 then return 'at boiling point'
+    if temp < 100 then return 'below boiling point'
+if unit = 'K'
+    if temp > 373 then return 'above boiling point'
+    if temp === 373 then return 'at boiling point'
+    if temp < 373 then return 'below boiling point'
+if unit = 'F'
+    if temp > 212 then return 'above boiling point'
+    if temp === 212 then return 'at boiling point'
+    if temp < 212 then return 'below boiling point'
+*/
```

Comment:
Wow!!! 💯 Amazing pseudocode and the function is just as impressive! You account for the inputs you may receive very well. 👏🏽